yt/visualization/_handlers.py

```diff
@@ -374,6 +353,54 @@ def get_norm(self, data: np.ndarray, *args, **kw) -> Normalize:
 
         return norm_type(*args, **kw)
 
+    def _guess_linthresh(self, finite_plot_data):
+        # finite_plot_data is the ImageArray or ColorbarHandler data, already
+        # filtered to be finite values
+
+        # get the extrema for the negative and positive values separately
+        # neg_min -> neg_max -> 0 -> pos_min -> pos_max
+        def get_minmax(data):
+            if len(data) > 0:
+                return np.nanmin(data), np.nanmax(data)
+            return None, None
+
+        pos_min, pos_max = get_minmax(finite_plot_data[finite_plot_data > 0])
+        neg_min, neg_max = get_minmax(finite_plot_data[finite_plot_data < 0])
+
+        has_pos = pos_min is not None
+        has_neg = neg_min is not None
+
+        # the starting guess is the absolute value of the point closest to 0
+        # (remember: neg_max is closer to 0 than neg_min)
+        if has_pos and has_neg:
+            linthresh = np.min((np.abs(neg_max), pos_min))
+        elif has_pos:
+            linthresh = pos_min
+        elif has_neg:
+            linthresh = np.abs(neg_max)
+        else:
+            # this condition should handled before here in get_norm
+            raise RuntimeError("No negative or positive values")
+
+        log10_linthresh = np.log10(linthresh)
+
+        # if either the pos or neg ranges exceed cutoff_sigdigs, then
+        # linthresh is shifted to decrease the range to avoid floating point
+        # precision errors in the normalization.
+        cutoff_sigdigs = 15.0  # max allowable range in significant digits
```

Comment:
An integer feels more appropriate
```suggestion
        cutoff_sigdigs = 15  # max allowable range in significant digits
```